### PR TITLE
`matchFieldName` and `matchFieldOptionValue` options

### LINKED ISSUE #25: add `matchFieldName` and `matchFieldOptionValue` options for custom matching of field names and field option values
### Please avoid duplicates

- [X] I checked [all open feature requests](https://github.com/gr2m/github-project/issues?q=is%3Aissue+is%3Aopen+label%3Afeature) and none of them matched my problem.

### What’s missing?

There is currently no way to retrieve all fields and field options as they currently exist in a Project

### Why?

Custom mapping of field names and field option values. 

Right now field names and option values are compared explicitly, including capitalization. A common use case is to lowercase / remove white space / remove punctuation / remove emoji before comparison

Say I instantiate a project with

```js
const project = new Project({
  org,
  number,
  token,
  fields: {
    type: "Type",
  },
});
```

and the project does not have field with the name `"Type"`, but it has a field with the name `"🗂️ Type"`, then retrieving or updating items will fail.

An alternative approach might be to add options:

- `matchFieldName`: defaults to `(projectFieldName, userFieldName) => projectFieldName === userFieldName`. `projectFieldName` and  `userFieldName` are always lower cased as Projects to not permit the same field names with different capitalization only. E.g. having a field "Project" and a field "project" is not possible.
- `matchFieldOptionValue`: defaults to `(fieldOptionValue, userValue) => fieldOptionValue === userValue`

They be overwritten to apply whatever logic a project wants to use for matching

### Alternatives you tried

Using custom code to retrieve all project fields and their field options

### Would you be interested in contributing the feature?

- [X] yes

### PR DESCRIPTION
**Pull request by gr2m**
[Preview of documentation](https://github.com/gr2m/github-project/blob/25-add-matchfieldname-and-matchfieldoptionvalue-options-for-custom-matching-of-field-names-and-field-option-values/README.md#constructor). closes #25

### CODE REVIEW

**Comment by mikesurowiec** [comment]
Looks great!

**Review by gr2m — COMMENTED** [review]
Now that we offer custom matching functions, there is a chance that multiple fields / field options can be matched, which I'd say should throw a helpful error. But we can address this in a follow up issue, I'd say it's an edge case and shouldn't block this feature from being released

File: README.md [hunk 1]
```diff
@@ -172,7 +172,59 @@ const project = new GitHubProject(options);
...
+        <code>Function</code>
+      </td>
+      <td>
+
+Customize how field names are matched with the values provided in `options.fields`. The function accepts two arguments:
+
+1. `projectFieldName`
+2. `userFieldName`
+
+Both are strings. Both arguments are lower-cased and trimmed before passed to the function. The function must return `true` or `false`.
```
Comment (mikesurowiec) [inline README.md:193]: This is because GitHub lower-case + trims the field names, right? As in the user can't possibly attempt a case-sensitive match

**Review by mikesurowiec — APPROVED** [review]
👏  awesome stuff!

File: README.md (hunk 1 above)
Comment (gr2m) [reply README.md:193]: Yes, exactly

Comment (gr2m) [reply README.md:193]: Oh which made me think to check if option values are trimmed by GitHub, and indeed they are, I'll adjust the code and then merge it

**Pull request merged**